src/oscar/apps/dashboard/vouchers/views.py

```diff
@@ -95,39 +108,20 @@ def get_context_data(self, **kwargs):
         return ctx
 
     def get_initial(self):
-        return dict(
-            exclusive=True
-        )
+        return {
+            'start_datetime': timezone.now(),
+        }
 
     @transaction.atomic()
     def form_valid(self, form):
-        # Create offer and benefit
-        condition = Condition.objects.create(
-            range=form.cleaned_data['benefit_range'],
-            type=Condition.COUNT,
-            value=1
-        )
-        benefit = Benefit.objects.create(
-            range=form.cleaned_data['benefit_range'],
-            type=form.cleaned_data['benefit_type'],
-            value=form.cleaned_data['benefit_value']
-        )
-        name = form.cleaned_data['name']
-        offer = ConditionalOffer.objects.create(
-            name=get_offer_name(name),
-            offer_type=ConditionalOffer.VOUCHER,
-            benefit=benefit,
-            condition=condition,
-            exclusive=form.cleaned_data['exclusive'],
-        )
         voucher = Voucher.objects.create(
```

Comment:
Is it possible to subclass `django.views.generic.edit.CreateView` here now that we're not creating offers on the fly? Feels like that would do the object creation for us, plus it would handle the unique constraint checking that we are currently doing manually in `clean_name`, `clean_code` (we would still need to clean the code to uppercase, but would no longer need to check the uniqueness ourselves).

Reply:
Similarly subclassing `UpdateView` for `VoucherUpdateView` may help us cut out a lot of code there as well.